src/infrastructure/HostNetworkManager_linux.go

```diff
@@ -361,6 +378,129 @@ func (h *HostNetworkManager) AddrDelete(linkName string, addr net.IPNet) error {
+	if err != nil {
+		return domain.HostNetworkTrafficRule{}, errors.Internal.Wrap(err, "failed to create traffic rule")
+	}
+	return rule, nil
+}
+
+//DeleteTrafficRule Delete netfilter traffic rule in specified table
+//
+//Params:
+//	table - table to delete a rule
+//	rule - rule entity
+//Return:
+//	error - if an error occurs, otherwise nil
+func (h *HostNetworkManager) DeleteTrafficRule(table string, rule domain.HostNetworkTrafficRule) error {
+	rulespec := h.parseRule(rule)
+	exist, err := h.iptables.Exists(table, rule.Chain, rulespec...)
+	if err != nil {
+		return errors.Internal.Wrap(err, "failed to check existence of traffic rule")
+	}
+	if exist {
+		err = h.iptables.Delete(table, rule.Chain, rulespec...)
+		if err != nil {
+			return errors.Internal.Wrap(err, "failed to delete traffic rule")
+		}
+		return nil
+	}
+	return errors.NotFound.New("traffic rule not found")
+}
+
+//GetChainRules Get selected netfilter chain rules at specified table
+//
+//Params:
+//	table - table to get a rules
+//	chain - chain where we get the rules
+//Return:
+//	[]domain.HostNetworkTrafficRule - slice of rules
+//	error - if an error occurs, otherwise nil
+func (h *HostNetworkManager) GetChainRules(table string, chain string) ([]domain.HostNetworkTrafficRule, error) {
+	var rules []domain.HostNetworkTrafficRule
+
+	list, err := h.iptables.Stats(table, chain)
+	if err != nil {
+		return nil, errors.Internal.Wrap(err, "failed to get list of traffic rules")
+	}
+	for _, l := range list {
+		stat, err := h.iptables.ParseStat(l)
+		if err != nil {
+			return nil, errors.Internal.Wrap(err, "failed to parse traffic rule to stat struct")
+		}
+		rule := &domain.HostNetworkTrafficRule{Chain: chain}
+		mappers.MapStatToTrafficRule(stat, rule)
+		rules = append(rules, *rule)
+	}
+	return rules, err
+}
+
+//GetTableRules Get specified netfilter table rules
+//
+//Params:
+//	table - table to get a rules
+//Return:
+//	[]domain.HostNetworkTrafficRule - slice of rules
+//	error - if an error occurs, otherwise nil
+func (h *HostNetworkManager) GetTableRules(table string) ([]domain.HostNetworkTrafficRule, error) {
+	var rules []domain.HostNetworkTrafficRule
+
+	chains, err := h.iptables.ListChains(table)
+	if err != nil {
+		return nil, errors.Internal.Wrap(err, "failed to get list table chains")
+	}
+	for _, chain := range chains {
+		list, err := h.iptables.Stats(table, chain)
+		if err != nil {
+			return nil, errors.Internal.Wrap(err, "failed to get list of traffic rules")
+		}
+		for _, l := range list {
+			stat, err := h.iptables.ParseStat(h.trimExclamationMarkInStat(l))
+			if err != nil {
+				return nil, errors.Internal.Wrap(err, "failed to parse traffic rule to stat struct")
+			}
+			rule := &domain.HostNetworkTrafficRule{Chain: chain}
+			mappers.MapStatToTrafficRule(stat, rule)
+			rules = append(rules, *rule)
+		}
+	}
+	return rules, nil
+}
+
+func (h *HostNetworkManager) trimExclamationMarkInStat(slice []string) (out []string) {
```

Comment:
Move this method higher than the place where it is used.

Reply:
done